pkg/pgmodel/cache/cache.go

```diff
@@ -13,12 +14,16 @@ import (
 
 const (
 	DefaultMetricCacheSize = 10000
+
+	nullSchemaName = "__null__"
+	fieldSeparator = "*"
```

Comment:
is there a reason `*` can't be a valid schema or table name? How do we protect against that?

Reply:
I believe we don't have to, since Postgres doesn't allow it.

> SQL identifiers and key words must begin with a letter (a-z, but also letters with diacritical marks and non-Latin letters) or an underscore (_). Subsequent characters in an identifier or key word can be letters, underscores, digits (0-9), or dollar signs ($). 

https://www.postgresql.org/docs/current/sql-syntax-lexical.html

I guess the only thing that could contain it is the metric name (which is very unlikely) and should not break anything here.

Reply:
But there is a second kind of identifier 

> There is a second kind of identifier: the delimited identifier or quoted identifier. It is formed by enclosing an arbitrary sequence of characters in double-quotes ("). A delimited identifier is always an identifier, never a key word. So "select" could be used to refer to a column or table named “select”, whereas an unquoted select would be taken as a key word and would therefore provoke a parse error when used where a table or column name is expected. The example can be written with quoted identifiers like this:

When they say arbitrary characters they mean it - full unicode - I've seen emoji-named tables


I think the only really safe way to do this is to have length-delimited fields not fields with separators.

Reply:
If `fieldSeparator` is just to separate fields, why not use something that is more common in separating values like `;` or even `,`?